pkg/controller/llmisvc/workload_single_node.go

```diff
@@ -182,9 +185,13 @@ func (r *LLMInferenceServiceReconciler) expectedPrefillMainDeployment(ctx contex
 		d.Spec.Template.Spec = *llmSvc.Spec.Prefill.Template.DeepCopy()
 	}
 
+	if err := r.attachModelArtifacts(llmSvc, &d.Spec.Template.Spec, storageConfig); err != nil {
```

Comment:
As a follow-up I wonder if we shouldn't move `attachModelArtifacts` validation parts to the webhook so that it will simply assume the data is valid. WDYT?